common/web/keyboard-processor/build.sh

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env bash
+#
+# Compile KeymanWeb's 'keyboard-processor' module, one of the components of Web's 'core' module.
+#
+set -eu
+
+## START STANDARD BUILD SCRIPT INCLUDE
+# adjust relative paths as necessary
+THIS_SCRIPT="$(greadlink -f "${BASH_SOURCE[0]}" 2>/dev/null || readlink -f "${BASH_SOURCE[0]}")"
+. "$(dirname "$THIS_SCRIPT")/../../../resources/build/build-utils.sh"
+. "$KEYMAN_ROOT/resources/shellHelperFunctions.sh"
+## END STANDARD BUILD SCRIPT INCLUDE
+
+# This script runs from its own folder
+cd "$(dirname "$THIS_SCRIPT")"
+
+action=
+function action_failure() {
+  if [ -n "$action" ]; then
+    builder_report failure $action
+  fi
+}
+
+trap action_failure err
+
+################################ Main script ################################
+
+c1="${COLOR_BLUE:=<}"
+c0="${COLOR_RESET:=>}"
```

Comment:
Nice. I wonder if we can genericise this one day?

Reply:
I just ripped this out from the builder-utils `describe` function to make sure it matched the format used there.  We probably can, in some fashion.  I'll take a look at it while I see if I can generalize the `trap` metaphor.